src/cpl/nuopc/rof_comp_nuopc.F90

```diff
@@ -635,35 +646,38 @@ subroutine ModelAdvance(gcomp, rc)
     if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
     !--------------------------------
-    ! Determine if time to write restart
+    ! Determine if time to stop
```

Comment:
@ekluzek for your review:
I realize that this section may seem confusing...
I had to reorder the "stop" and "write restart" sections, putting "stop" first instead of second. This way the "write_restart" section has access to the nlend boolean that it needs.